src/strands/agent/conversation_manager/summarizing_conversation_manager.py

```
@@ -0,0 +1,192 @@
+            summarization_model: Model to use for generating summaries.
+                Required if enable_summarization is True.
+            summary_ratio: Ratio of messages to summarize vs keep when window is exceeded.
+                Value between 0.1 and 0.8. Defaults to 0.3 (summarize 30% of oldest messages).
+            preserve_recent_messages: Minimum number of recent messages to always keep.
+                Defaults to 10 messages.
+        """
+        super().__init__(window_size)
+        self.enable_summarization = enable_summarization
+        self.summarization_model = summarization_model
+        self.summary_ratio = max(0.1, min(0.8, summary_ratio))
+        self.preserve_recent_messages = preserve_recent_messages
+
+        if enable_summarization and summarization_model is None:
+            raise ValueError("summarization_model is required when enable_summarization is True")
+
+    def reduce_context(self, messages: Messages, e: Optional[Exception] = None) -> None:
+        """Reduce context using summarization if enabled, otherwise fall back to sliding window.
+
+        Args:
+            messages: The messages to reduce.
+                This list is modified in-place.
+            e: The exception that triggered the context reduction, if any.
+
+        Raises:
+            ContextWindowOverflowException: If the context cannot be reduced further.
+        """
+        if not self.enable_summarization or self.summarization_model is None:
+            # Fall back to standard sliding window behavior
+            super().reduce_context(messages, e)
+            return
+
+        try:
+            self._reduce_context_with_summarization(messages, e)
+        except Exception as summarization_error:
+            logger.warning("Summarization failed, falling back to sliding window: %s", summarization_error)
+            # Fall back to sliding window if summarization fails
```

Lets not fall back to sliding window here. The `SlidingWindowConversationManager` completely removes context, where the summarization is meant to retain that context. We can just throw an exception here instead.

---

Agreed, removed the fallback and raising the exception.